File: 전예진/Hash/prob3.py
# Prob3. 전화번호 목록
# 알게된 점 - 문자열1.startswith('문자열2')을 사용하면, 어떤 문자열1이 문자열2로 시작하는지 확인함 

def solution(phone_book):
    answer = True
    
    phone_book.sort()
    for i in range(0, len(phone_book)-1):
        if phone_book[i+1].startswith(phone_book[i]):
            answer = False
            
    return answer

File: 이주은/Sort/prob3.java
//[프로그래머스] H-Index(https://school.programmers.co.kr/learn/courses/30/lessons/42747#)

import java.util.Arrays;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        
        Arrays.sort(citations);
        for (int i = 0; i < citations.length; i++) {
            int h = citations.length - i; //h번 이상 인용된 논문 수
            if (citations[i] >= h) { //h번 인용된 논문이면
                answer = h;
                break;
            }
        }
        return answer;
    }
}
